don't sys.exit() on missing modules... causes sr3 features not to work.

```
(rhat8venv) edcm-dirt-rhel8-1% sr3 features
2023-12-14 15:25:38,267 149995 [ERROR] sarracenia.credentials isValid ssh_keyfile not found: /home/sarra/.ssh/id_rsa
2023-12-14 15:25:38,267 149995 [ERROR] sarracenia.credentials _parse bad credential 2 (sftp://envican@opgftpanon.opg.com/ ssh_keyfile=/home/sarra/.ssh/id_rsa
)
2023-12-14 15:25:38,329 149995 [ERROR] sarracenia.credentials isValid credential not found
2023-12-14 15:25:38,329 149995 [CRITICAL] root _validate_urlstr bad credential amqps://OPG@ddsr-dev.cmc.ec.gc.ca/
2023-12-14 15:25:39,716 149995 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'scheduled.wiski', 'log']
2023-12-14 15:25:39,750 149995 [INFO] sarracenia.flowcb.scheduled update_appointments for 2023-12-14 20:25:39.750801+00:00: []
2023-12-14 15:25:39,751 149995 [INFO] sarracenia.flowcb.scheduled.wiski __init__ main_url: https://kiwis.opg.com/KiWIS/KiWIS timeseries_length=1 day, 0:00:00, writing to /tmp/opg_wiski
2023-12-14 15:25:39,752 149995 [INFO] sarracenia.flowcb.log __init__ flow initialized with: logEvents: {'after_work', 'after_post', 'post', 'on_housekeeping', 'after_accept'},  logMessageDump: False
2023-12-14 15:25:39,752 149995 [WARNING] sarracenia.flow.poll __init__ post_exchange: xs_PROVINCIAL_ab_airnow_poll is different from exchange: ['xs_PROVINCIAL_ab_airnow_poll']. The settings need for multiple instances to share a poll.
2023-12-14 15:25:39,752 149995 [INFO] sarracenia.flow.poll __init__ adding poll plugin, because missing from: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log']
2023-12-14 15:25:39,752 149995 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.post.message.Message', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log', 'sarracenia.flowcb.poll.Poll']
2023-12-14 15:25:39,759 149995 [INFO] sarracenia.flowcb.nodupe.disk __init__ time_to_live=25200,
2023-12-14 15:25:39,759 149995 [INFO] sarracenia.flowcb.log __init__ poll initialized with: logEvents: {'after_work', 'after_post', 'post', 'on_housekeeping', 'after_accept'},  logMessageDump: False
2023-12-14 15:25:39,763 149995 [ERROR] sarracenia.config check_undeclared_options poll/airnow_ab.conf:44 undeclared option: to
2023-12-14 15:25:39,763 149995 [WARNING] sarracenia.flow.poll __init__ post_exchange: xs_feed_QuebecAQ is different from exchange: ['xs_feed_QuebecAQ']. The settings need for multiple instances to share a poll.
2023-12-14 15:25:39,763 149995 [INFO] sarracenia.flow.poll __init__ adding poll plugin, because missing from: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log']
2023-12-14 15:25:39,763 149995 [CRITICAL] sarracenia.flow.poll __init__ attempting to configure an FTP poll pollUrl=ftp://ftp.mddep.gouv.qc.ca/, but missing python modules: dateparser pytz
(rhat8venv) edcm-dirt-rhel8-1%
```

```

(rhat8venv) edcm-dirt-rhel8-1% sr3 features
2023-12-14 15:27:26,982 150005 [ERROR] sarracenia.credentials isValid ssh_keyfile not found: /home/sarra/.ssh/id_rsa
2023-12-14 15:27:26,982 150005 [ERROR] sarracenia.credentials _parse bad credential 2 (sftp://envican@opgftpanon.opg.com/ ssh_keyfile=/home/sarra/.ssh/id_rsa
)
2023-12-14 15:27:27,041 150005 [ERROR] sarracenia.credentials isValid credential not found
2023-12-14 15:27:27,041 150005 [CRITICAL] root _validate_urlstr bad credential amqps://OPG@ddsr-dev.cmc.ec.gc.ca/
2023-12-14 15:27:28,391 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'scheduled.wiski', 'log']
2023-12-14 15:27:28,429 150005 [INFO] sarracenia.flowcb.scheduled update_appointments for 2023-12-14 20:27:28.429499+00:00: []
2023-12-14 15:27:28,429 150005 [INFO] sarracenia.flowcb.scheduled.wiski __init__ main_url: https://kiwis.opg.com/KiWIS/KiWIS timeseries_length=1 day, 0:00:00, writing to /tmp/opg_wiski
2023-12-14 15:27:28,431 150005 [INFO] sarracenia.flowcb.log __init__ flow initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False
2023-12-14 15:27:28,431 150005 [WARNING] sarracenia.flow.poll __init__ post_exchange: xs_PROVINCIAL_ab_airnow_poll is different from exchange: ['xs_PROVINCIAL_ab_airnow_poll']. The settings need for multiple instances to share a poll.
2023-12-14 15:27:28,431 150005 [INFO] sarracenia.flow.poll __init__ adding poll plugin, because missing from: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log']
2023-12-14 15:27:28,431 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.post.message.Message', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log', 'sarracenia.flowcb.poll.Poll']
2023-12-14 15:27:28,440 150005 [INFO] sarracenia.flowcb.nodupe.disk __init__ time_to_live=25200,
2023-12-14 15:27:28,440 150005 [INFO] sarracenia.flowcb.log __init__ poll initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False
2023-12-14 15:27:28,445 150005 [ERROR] sarracenia.config check_undeclared_options poll/airnow_ab.conf:44 undeclared option: to
2023-12-14 15:27:28,445 150005 [WARNING] sarracenia.flow.poll __init__ post_exchange: xs_feed_QuebecAQ is different from exchange: ['xs_feed_QuebecAQ']. The settings need for multiple instances to share a poll.
2023-12-14 15:27:28,445 150005 [INFO] sarracenia.flow.poll __init__ adding poll plugin, because missing from: ['sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log']
2023-12-14 15:27:28,446 150005 [CRITICAL] sarracenia.flow.poll __init__ attempting to configure an FTP poll pollUrl=ftp://ftp.mddep.gouv.qc.ca/, but missing python modules: dateparser pytz
2023-12-14 15:27:28,446 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.post.message.Message', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'sarracenia.flowcb.nodupe.disk.Disk', 'log', 'sarracenia.flowcb.poll.Poll']
2023-12-14 15:27:28,447 150005 [DEBUG] sarracenia.diskqueue __init__ name=work_retry_00 logLevel=debug
2023-12-14 15:27:28,447 150005 [DEBUG] sarracenia.diskqueue __init__  post_retry_000 __init__
2023-12-14 15:27:28,447 150005 [DEBUG] sarracenia.diskqueue __init__ name=post_retry_000 logLevel=debug
2023-12-14 15:27:28,448 150005 [DEBUG] sarracenia.flowcb.retry __init__ logLevel=debug
2023-12-14 15:27:28,449 150005 [DEBUG] sarracenia.flowcb.nodupe.disk __init__ NoDupe init
2023-12-14 15:27:28,449 150005 [INFO] sarracenia.flowcb.nodupe.disk __init__ time_to_live=25200,
2023-12-14 15:27:28,449 150005 [INFO] sarracenia.flowcb.log __init__ poll initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False
2023-12-14 15:27:28,450 150005 [DEBUG] sarracenia.transfer __init__ class=Ftp , subclasses=[<class 'sarracenia.transfer.file.File'>, <class 'sarracenia.transfer.ftp.Ftp'>, <class 'sarracenia.transfer.https.Https'>, <class 'sarracenia.transfer.sftp.Sftp'>]
2023-12-14 15:27:28,450 150005 [DEBUG] sarracenia.transfer.ftp __init__ sr_ftp __init__
2023-12-14 15:27:28,450 150005 [DEBUG] sarracenia.flow loadCallbacks complete
2023-12-14 15:27:28,450 150005 [ERROR] sarracenia.config check_undeclared_options poll/QC_AQ.conf:21 undeclared option: to
2023-12-14 15:27:28,450 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.post.message.Message', 'sarracenia.flowcb.gather.file.File', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'log']
2023-12-14 15:27:28,455 150005 [CRITICAL] sarracenia.flowcb.gather.file __init__ watchdog module must be installed to watch directories
2023-12-14 15:27:28,455 150005 [DEBUG] sarracenia.flowcb.gather.file __init__ post used to be overwrite_defaults
2023-12-14 15:27:28,455 150005 [DEBUG] sarracenia.flowcb.retry __init__ sr_retry __init__
2023-12-14 15:27:28,456 150005 [DEBUG] sarracenia.diskqueue __init__  work_retry_00 __init__
2023-12-14 15:27:28,456 150005 [DEBUG] sarracenia.diskqueue __init__ name=work_retry_00 logLevel=debug
2023-12-14 15:27:28,456 150005 [DEBUG] sarracenia.diskqueue __init__  post_retry_000 __init__
2023-12-14 15:27:28,456 150005 [DEBUG] sarracenia.diskqueue __init__ name=post_retry_000 logLevel=debug
2023-12-14 15:27:28,456 150005 [DEBUG] sarracenia.flowcb.retry __init__ logLevel=debug
2023-12-14 15:27:28,457 150005 [INFO] sarracenia.flowcb.log __init__ post initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False
2023-12-14 15:27:28,457 150005 [DEBUG] sarracenia.flow loadCallbacks complete
2023-12-14 15:27:28,457 150005 [ERROR] sarracenia.config check_undeclared_options post/mqtt_hpfx.conf:7 undeclared option: nodupe
2023-12-14 15:27:28,457 150005 [ERROR] sarracenia.config check_undeclared_options post/mqtt_hpfx.conf:8 undeclared option: retry
2023-12-14 15:27:28,460 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.post.message.Message', 'sarracenia.flowcb.gather.message.Message', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'log']
2023-12-14 15:27:28,461 150005 [ERROR] sarracenia.moth pubFactory no broker specified
2023-12-14 15:27:28,463 150005 [DEBUG] sarracenia.flowcb.retry __init__ sr_retry __init__
2023-12-14 15:27:28,463 150005 [DEBUG] sarracenia.diskqueue __init__  work_retry_00 __init__
2023-12-14 15:27:28,464 150005 [INFO] sarracenia.flowcb.log __init__ shovel initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False
2023-12-14 15:27:28,468 150005 [INFO] sarracenia.flow loadCallbacks flowCallback plugins to load: ['sarracenia.flowcb.gather.message.Message', 'sarracenia.flowcb.retry.Retry', 'sarracenia.flowcb.housekeeping.resources.Resources', 'log']
2023-12-14 15:27:28,471 150005 [INFO] sarracenia.flowcb.log __init__ subscribe initialized with: logEvents: {'after_accept', 'after_post', 'post', 'on_housekeeping', 'after_work'},  logMessageDump: False

Status:    feature:   python imports:      Description:
Installed  amqp       amqp                 can connect to rabbitmq brokers
Installed  appdirs    appdirs              place configuration and state files appropriately for platform (windows/mac/linux)
Installed  filetypes  magic                able to set content headers
Absent     ftppoll    dateparser,pytz      not able to poll with ftp
Installed  humanize   humanize,humanfriendly humans numbers that are easier to read.
Absent     mqtt       paho.mqtt.client     cannot connect to mqtt brokers
Installed  process    psutil               can monitor, start, stop processes:  Sr3 CLI should basically work
Absent     redis      redis,redis_lock     cannot use redis implementations of retry and nodupe
Installed  retry      jsonpickle           can write messages to local queues to retry failed publishes/sends/downloads
Installed  sftp       paramiko             can use sftp or ssh based services
Installed  vip        netifaces            able to use the vip option for high availability clustering
Absent     watch      watchdog             cannot watch directories
Absent     xattr      xattr                unable to store additional file metadata in extended attributes
MISSING    wiski      kiwis_pie            cannot poll sites that provide data using WISKI API

 state dir: /net/local/home/silvap/.cache/sr3
 config dir: /net/local/home/silvap/.config/sr3

(rhat8venv) edcm-dirt-rhel8-1%

```

with fix, it now prints the features and shows the missing modules.

--- a/sarracenia/flow/poll.py
+++ b/sarracenia/flow/poll.py
@@ -77,7 +77,6 @@ class Poll(Flow):
         if not features['ftppoll']['present']:
             if hasattr( self.o, 'pollUrl' ) and ( self.o.pollUrl.startswith('ftp') ):
                 logger.critical( f"attempting to configure an FTP poll pollUrl={self.o.pollUrl}, but missing python modules: {' '.join(features['ftppoll']['modules_needed'])}" )
-                sys.exit(1)
 
     def do(self):
         """

--- a/sarracenia/moth/__init__.py
+++ b/sarracenia/moth/__init__.py
@@ -44,7 +44,6 @@ def ProtocolPresent(p) -> bool:
     else:
         logger.critical( f"Protocol scheme {p} unsupported for communications with message brokers" )
 
-    sys.exit(1)
     return False
 
 

